File: src/main/java/org/taktik/icure/entities/embed/FlatRateTarification.java
package org.taktik.icure.entities.embed;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;

import java.io.Serializable;
import java.util.Map;
import java.util.Set;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class FlatRateTarification implements Serializable {

    protected String code;
    protected FlatRateType flatRateType;
    protected java.util.Map<String, String> label;
    protected Set<Valorisation> valorisations;

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public Map<String, String> getLabel() {
        return label;
    }

    public void setLabel(Map<String, String> label) {
        this.label = label;
    }

    public Set<Valorisation> getValorisations() {
        return valorisations;
    }

    public void setValorisations(Set<Valorisation> valorisations) {
        this.valorisations = valorisations;
    }

    public FlatRateType getFlatRateType() { return flatRateType; }

    public void setFlatRateType(FlatRateType flatRateType) { this.flatRateType = flatRateType; }
}

File: src/main/kotlin/org/taktik/icure/entities/embed/FlatRateTarification.kt
package org.taktik.icure.entities.embed

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.github.pozo.KotlinBuilder
import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@KotlinBuilder
data class FlatRateTarification(
        val code: String? = null,
        val flatRateType: FlatRateType? = null,
        val label: Map<String, String>? = null,
        val valorisations: List<Valorisation> = listOf(),
        override val encryptedSelf: String? = null
) : Encrypted, Serializable
